cache/sampleStore.js

```diff
@@ -81,10 +103,12 @@ function eradicate() {
  */
 function removeNullsAndStringifyArrays(obj, arrayFields) {
   Object.keys(obj).forEach((key) => {
-    if (arrayFields.includes(key)) {
-      obj[key] = JSON.stringify(obj[key]);
-    } else if (obj[key] === null) {
+    // delete null fields and empty arrays, then stringify array fields
+    if (obj[key] === null ||
+     (Array.isArray(obj[key]) && obj[key].length === 0)) {
```

Comment:
deleting empty arrays would be breaking the current API contract. Anything that is not a null is allowed.